### ISSUE TITLE
[Accessibility]: Saved Searches ARC toolkit issues

### ISSUE THREAD

**Issue by limitedmage** [sourcegraph/sourcegraph#35169]
### Audit type

ARC Toolkit

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34658

### Problem description

- SVG icon in "+ Add saved search" should be aria-hidden
<img width="192" alt="image" src="https://user-images.githubusercontent.com/206864/167493971-8b366358-276d-4a6c-a5fd-2b481dfe9a79.png">

- SVG icons in saved search list (saved search icon, settings icon, delete icon) should be aria-hidden
![CleanShot 2022-05-09 at 16 40 46@2x](https://user-images.githubusercontent.com/206864/167494188-20e541ce-d3d7-44ad-ac81-0618d17c0f9c.png)

- Header on user/org profile page (user/org name) should be h2, not h1
<img width="367" alt="image" src="https://user-images.githubusercontent.com/206864/167494250-b8621b7d-93b2-472d-ab0a-3798cc294785.png">

- Header on saved searches pages ("Saved searches" on list page and "Add saved search"/"Manage saved search" on create/edit page) should be h3, not h2
<img width="411" alt="image" src="https://user-images.githubusercontent.com/206864/167494310-d8269f7f-afcb-4527-8377-9b6ae4901351.png">
<img width="497" alt="image" src="https://user-images.githubusercontent.com/206864/167494342-532d7804-2fcc-4bb8-9909-92af44295a9c.png">

### Expected behavior

Issues fixed as described above

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [X] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by GitStart. I have assigned the label 'gitstart' and added the assignee 'gitstart-sourcegraph'.
